### ISSUE TITLE
R2DfgFrigate: add antimissile & antiplasma capabilities

### ISSUE THREAD

**Issue by KajiTetsushi** [KajiTetsushi/HomeworldSDL#8]
Like #7, this is to introduce some usefulness to the ship. Ideally, the defense field frigate should have more defensive power than the defense fighter.

**Comment by Talthos** [comment]
Might be easier to have the DFG Frigate take cues from its incarnation in Cataclysm:

- Defense field is 'always on', but simply deflects a fixed percentage of incoming bullet fire

- Multiple DFG Frigates in close formation will stack the overlapping defense fields, and deflect a greater number of bullets than just 1 DFG Frigate

- Defense field will never block ion cannon beams, missiles, nor mines

**Comment by KajiTetsushi** [comment]
> Defense field is 'always on'

@Talthos Nah, this wasn't about adding an "on" state to it. I think that it's fine to be always on as it is simply because of the micromanagement involved in the heat of (multiplayer) battle. I'm not necessarily taking cues from Homeworld2 here.

> but simply deflects a fixed percentage of incoming bullet fire

Always has been this way. Nothing will change here.

> overlapping defense fields

Again, always has been, I think?

> never block ion cannon beams, missiles nor mines

Ion cannon beams, I agree but missiles and mines, I'm not sure. It'd add more Defense Field Frigate (DFG) viability to have a fraction of the amount of incoming missiles be shut down, like a modern-day missile jammer. My impression of the DFG is that is that it's not going to be used a lot if it doesn't deflect anything other than kinetic ammo. Correct me if I'm wrong. Opening this up for modders (not the original flavor of gameplay) is basically why I thought of this.

**Comment by Talthos** [comment]
> I think that it's fine to be always on as it is simply because of the micromanagement involved in the heat of (multiplayer) battle.

In Homeworld 1, the DFG Frigate's defense shield is a manually toggled ability, and must recharge afterwards.

In Cataclysm, it was changed to a passive shield that does not require any manual toggling; it is 'always on.'

That's what I was getting at.

> Opening this up for modders (not the original flavor of gameplay) is basically why I thought of this.

Ah. That works just as well.

**Comment by KajiTetsushi** [comment]
> In Homeworld 1, the DFG Frigate's defense shield is a manually toggled ability, and must recharge afterwards.

Not quite. The Homeworld 1999 DFG has always been a passive ability. Unless we see some kind of demand to open up a configuration that would convert the defense field into an active toggle ability, we wouldn't be changing that.

To prove this in the code as of `main` branch at this time: https://github.com/KajiTetsushi/HomeworldSDL/blob/d93dd9230a0403a6e96a250e384249399bfd3176/src/Ships/DFGFrigate.c#L169-L189

Line 180 is `NULL`. See that? In [`CustShipHeader`](https://github.com/KajiTetsushi/HomeworldSDL/blob/main/src/Game/SpaceObj.h#L522-L542) (`CustShipAttackPassive`), that line is meant to indicate that the ship has an ability toggle. If there's a callback function assigned to it instead, **then** you have an ability toggle.

Or better yet, let's take a look at how `DFGFrigate` is imperatively configured. Check out **mouse.c**. 
https://github.com/KajiTetsushi/HomeworldSDL/blob/d93dd9230a0403a6e96a250e384249399bfd3176/src/SDL/mouse.c#L1403-L1409
https://github.com/KajiTetsushi/HomeworldSDL/blob/d93dd9230a0403a6e96a250e384249399bfd3176/src/SDL/mouse.c#L1331-L1356

Notice that none of these lines mention `DFGFrigate`? If it's not listed, then it's not a toggle ability. It's passive.
